File: 1935. Maximum Number of Words You Can Type.cpp
int canBeTypedWords(string text, string brokenLetters) {
        unordered_set<char> broken(brokenLetters.begin(), brokenLetters.end());
        int ans = 0;

        bool valid = true;
        for(int i = 0; i < text.length(); i++){
            if(text[i] == ' '){
                if(valid)
                    ans++;

                valid = true;

            }else if (broken.count(text[i]))
                valid = false;
        }
        if(valid)
            ans++;

        return ans;
    }

File: 1935. Maximum Number of Words You Can Type.kt
class Solution {
    fun canBeTypedWords(text: String, brokenLetters: String): Int {
        val invalid = brokenLetters.toSet();
        var ans = 0;
        var valid = true;
        for(c in text){
            if(c == ' '){
                if(valid) ans++

                valid = true
            }else if(c in invalid){
                valid = false
            }
        }

        if(valid) ans++

        return ans
    }
}

---------------------------

class Solution {
    fun canBeTypedWords(text: String, brokenLetters: String): Int {

        return text.split(" ").count{word -> word.none { it in brokenLetters }}
    }
}
